db/migrate/20230526085258_address_block_snapshot.rb

```diff
@@ -0,0 +1,12 @@
+class AddressBlockSnapshot < ActiveRecord::Migration[7.0]
+  def change
+    create_table :address_block_snapshots do |t|
+      t.belongs_to :address
+      t.belongs_to :block
+      t.bigint :block_number
+      t.jsonb :info
```

Comment:
We will save changes and final state here, so we must distinguish  these two field
I suggest rename `info` to `final_state` here, which means the states after change